aws/sqs_event_manager/sqs_event_manager/app.py

```diff
@@ -0,0 +1,118 @@
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with mash.  If not, see <http://www.gnu.org/licenses/>
+#
+"""
+Lambda to handle SQS messages from from
+AWS Metering/Entitlement Marketplace API's
+"""
+import json
+import logging
+
+from sqs_event_manager.queue import delete_message
+from sqs_event_manager.message import AWSSNSMessage
+
+logger = logging.getLogger()
+
+
+def lambda_handler(event, context):
+    """
+    Expects event type matching the SNS topic
+
+    {
+        'records': [
+            {
+                'message_id': 'str',
+                'receipt_handle': 'str',
+                'body': {
+                    'Type' : 'str',
+                    'MessageId : 'str',
+                    'TopicArn' : 'str',
+                    'Message' : {
+                        'action': 'str',
+                        'customer-identifier': 'str',
+                        'product-code': 'str',
+                    },
+                    'Timestamp' : 'str',
+                    'SignatureVersion' : 'str',
+                    'Signature' : 'str',
+                    'SigningCertURL' : 'str',
+                    'UnsubscribeURL' : 'str'
+                },
+                'attributes': {
+                    'ApproximateReceiveCount': 'str',
+                    'SentTimestamp': 'str',
+                    'SenderId': 'str',
+                    'ApproximateFirstReceiveTimestamp': 'str'
+                },
+                'messageAttributes': {},
+                'md5OfBody': 'str',
+                'eventSource': 'aws:sqs',
+                'eventSourceARN': 'arn:aws:sqs:us-east-1:111122223333:my-queue',
+                'awsRegion': 'us-east-1'
+            }
+        ]
+    }
+    """
+    batch_item_failures = []
+    sqs_batch_response = {}
+
+    try:
+        for message in event['Records']:
+            process_message(message, batch_item_failures)
+    except Exception as error:
+        return json.dumps(
+            error_response(500, f'{type(error).__name__}: {error}')
+        )
+
+    sqs_batch_response['batchItemFailures'] = batch_item_failures
+    return json.dumps(sqs_batch_response)
+
+
+def process_message(record: dict, batch_item_failures: dict):
+    """
+    Handle message received from SQS queue
+
+    Ensure the message is proper format. Get the customer entitlements and
+    send the information to the SCC service.
+    """
+    try:
+        message = AWSSNSMessage(record)
+
+        if not message.action:
+            logger.warning(f'Recieved an unknown message: {json.dumps(record)}')
+        elif message.action == 'entitlement-updated':
+            # Do we notify SCC to resolve customer API or do that here and
+            # send an update to SCC?
+            logger.debug(
```

Comment:
logger.debug will not land in CloudWatch even if you set the loglevel. I suggest to use logger.error and logger.info only